### PR TITLE
[KODOMAMO-422] Enable navigation to system home via plugin

### PR DESCRIPTION
**Pull request by abhishek10012001**
Closes KODOMAMO-422

Adding native implementation to move to home screen.

### CODE REVIEW

File: example/android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -1,28 +1,39 @@
...
-    <application android:label="flutter_overlay_window_example" android:name="${applicationName}" android:icon="@mipmap/ic_launcher">
+    <application android:label="flutter_overlay_window_example" android:name="${applicationName}"
+        android:icon="@mipmap/ic_launcher">
 
-        <service android:name="flutter.overlay.window.flutter_overlay_window.OverlayService" android:exported="false" />
+        <service android:name="flutter.overlay.window.flutter_overlay_window.OverlayService"
+            android:exported="false" />
 
-        <service android:name="slayer.accessibility.service.flutter_accessibility_service.AccessibilityListener" android:permission="android.permission.BIND_ACCESSIBILITY_SERVICE" android:exported="false">
+        <service
```
Comment (abhishek10012001) [inline example/android/app/src/main/AndroidManifest.xml:15]: Why this file has been formatted like this?

File: android/src/main/java/flutter/overlay/window/flutter_overlay_window/FlutterOverlayWindowPlugin.java [hunk 2]
```diff
@@ -109,7 +107,19 @@ public void onMethodCall(@NonNull MethodCall call, @NonNull Result result) {
                 result.success(true);
             }
             return;
-        } else {
+        } else if(call.method.equals("moveToHomeScreen")){
+            Intent goToSystemHome = new Intent(Intent.ACTION_MAIN);
+            goToSystemHome.addCategory(Intent.CATEGORY_HOME);
+            goToSystemHome.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+//            goToSystemHome.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
```
Comment (abhishek10012001) [inline android/src/main/java/flutter/overlay/window/flutter_overlay_window/FlutterOverlayWindowPlugin.java:114]: Please remove this line.

File: .fvm/fvm_config.json [hunk 3]
```diff
@@ -0,0 +1,4 @@
+{
+  "flutterSdkVersion": "3.7.11",
+  "flavors": {}
```
Comment (abhishek10012001) [inline .fvm/fvm_config.json:3]: Please remove these changes. Same for other files as well.

File: android/src/main/java/flutter/overlay/window/flutter_overlay_window/FlutterOverlayWindowPlugin.java (hunk 2 above)
Comment (sarthakydv) [reply android/src/main/java/flutter/overlay/window/flutter_overlay_window/FlutterOverlayWindowPlugin.java:114]: ok

File: example/android/app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (sarthakydv) [reply example/android/app/src/main/AndroidManifest.xml:15]: No, it is well formatted now

File: .fvm/fvm_config.json (hunk 3 above)
Comment (sarthakydv) [reply .fvm/fvm_config.json:3]: Why?

Comment (abhishek10012001) [reply .fvm/fvm_config.json:3]: It's a plugin and this seems to be developer settings oriented changes. No plugin has these changes as I have seen. That's why wondering should be keep it or remove it.

File: example/android/build.gradle [hunk 4]
```diff
@@ -26,6 +26,6 @@ subprojects {
     project.evaluationDependsOn(':app')
 }
 
-task clean(type: Delete) {
+tasks.register("clean", Delete) {
```
Comment (saurabhkumar8112) [inline example/android/build.gradle:29]: this change seem unnecessary

File: example/android/gradle.properties [hunk 5]
```diff
@@ -1,3 +1,9 @@
 org.gradle.jvmargs=-Xmx1536M
 android.useAndroidX=true
 android.enableJetifier=true
+org.gradle.jvmargs=-Xmx1536M \
+--add-exports=java.base/sun.nio.ch=ALL-UNNAMED \
```
Comment (saurabhkumar8112) [inline example/android/gradle.properties:5]: The --add-exports option is part of Java's module system (introduced in Java 9). It opens up the internal sun.nio.ch package from the java.base module to ALL-UNNAMED modules. This is not recommended in a production environment as it breaks encapsulation and could potentially harm the security and maintainability of your code.

File: example/android/gradle.properties [hunk 6]
```diff
@@ -1,3 +1,9 @@
...
+--add-opens=java.base/java.lang=ALL-UNNAMED \
```
Comment (saurabhkumar8112) [inline example/android/gradle.properties:6]: similar to above. Please remove these flags. I am 90% sure these aren't necessary

File: android/src/main/java/flutter/overlay/window/flutter_overlay_window/FlutterOverlayWindowPlugin.java [hunk 7]
```diff
@@ -100,16 +98,26 @@ public void onMethodCall(@NonNull MethodCall call, @NonNull Result result) {
...
                 result.success(true);
             }
-            return;
-        } else {
+        } else if(call.method.equals("moveToHomeScreen")){
+            Intent goToSystemHome = new Intent(Intent.ACTION_MAIN);
+            goToSystemHome.addCategory(Intent.CATEGORY_HOME);
+            goToSystemHome.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+            try {
+                startActivity(context,goToSystemHome,null);
```
[the remark this answers is not in the document]
Comment (sarthakydv) [reply android/src/main/java/flutter/overlay/window/flutter_overlay_window/FlutterOverlayWindowPlugin.java:114]: context is required

File: android/src/main/java/flutter/overlay/window/flutter_overlay_window/FlutterOverlayWindowPlugin.java [hunk 8]
```diff
@@ -100,16 +98,26 @@ public void onMethodCall(@NonNull MethodCall call, @NonNull Result result) {
...
         } else if (call.method.equals("closeOverlay")) {
             if (OverlayService.isRunning) {
                 final Intent i = new Intent(context, OverlayService.class);
                 i.putExtra(OverlayService.INTENT_EXTRA_IS_CLOSE_WINDOW, true);
                 context.startService(i);
                 result.success(true);
             }
-            return;
-        } else {
+        } else if(call.method.equals("moveToHomeScreen")){
```
[the remark this answers is not in the document]
Comment (sarthakydv) [reply android/src/main/java/flutter/overlay/window/flutter_overlay_window/FlutterOverlayWindowPlugin.java:109]: Yes, WIP

File: example/android/build.gradle (hunk 4 above)
Comment (sarthakydv) [reply example/android/build.gradle:29]: Auto gen

File: example/android/gradle.properties (hunk 5 above)
Comment (sarthakydv) [reply example/android/gradle.properties:5]: I could not find any other way to compile it

File: example/android/gradle.properties (hunk 6 above)
Comment (sarthakydv) [reply example/android/gradle.properties:6]: Removing the PR 🗿

File: .fvm/fvm_config.json (hunk 3 above)
Comment (sarthakydv) [reply .fvm/fvm_config.json:3]: we need to put it there in order to build in our framework

Comment (abhishek10012001) [reply .fvm/fvm_config.json:3]: I meant the same @saurabhkumar8112  that this should be standalone plugin.

**Pull request closed without merging**